integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/schema/management/ElasticsearchIndexSchemaManagerUpdateCustomMappingsIT.java

```diff
@@ -0,0 +1,185 @@
+
+		setupAndUpdateIndex( "no-overlapping.json" );
+		assertJsonEquals(
+				" { " +
+				"    '_source':{ " +
+				"       'enabled':false " +
+				"    }, " +
+				"    'properties':{ " +
+				"       '_entity_type':{ " +
+				"          'type':'keyword', " +
+				"          'index':false " +
+				"       }, " +
+				"       'bothField':{ " +
+				"          'type':'keyword', " +
+				"          'doc_values':false " +
+				"       }, " +
+				"       'bothObject':{ " +
+				"          'properties':{ " +
+				"             'bothNested':{ " +
+				"                'type':'keyword', " +
+				"                'doc_values':false " +
+				"             }, " +
+				"             'bothNestedObject':{ " +
+				"                'properties':{ " +
+				"                   'bothNestedNested':{ " +
+				"                      'type':'keyword', " +
+				"                      'doc_values':false " +
+				"                   }, " +
+				"                   'searchNestedNested':{ " +
+				"                      'type':'keyword', " +
+				"                      'doc_values':false " +
+				"                   } " +
+				"                } " +
+				"             }, " +
+				"             'searchNested':{ " +
+				"                'type':'keyword', " +
+				"                'doc_values':false " +
+				"             }, " +
+				"             'searchNestedObject':{ " +
+				"                'type':'object' " +
+				"             } " +
+				"          } " +
+				"       }, " +
+				"       'searchField':{ " +
+				"          'type':'keyword', " +
+				"          'doc_values':false " +
+				"       }, " +
+				"       'searchObject':{ " +
+				"          'type':'object' " +
+				"       }, " +
+				"       'userField':{ " +
+				"          'type':'keyword', " +
+				"          'norms':true " +
+				"       }, " +
+				"       'userObject':{ " +
+				"          'type':'object', " +
+				"          'dynamic':'true' " +
+				"       } " +
+				"    } " +
+				" } ",
+				elasticsearchClient.index( index.name() ).type().getMapping() );
+	}
+
+	@Test
+	public void illegalUpdate() {
+		elasticsearchClient.index( index.name() ).deleteAndCreate();
+		elasticsearchClient.index( index.name() ).type().putMapping(
+				" { " +
+				"    '_source':{ " +
+				"       'enabled':true " +
+				"    } " +
+				" } "
+		);
+
+		assertThatThrownBy( () -> setupAndUpdateIndex( "no-overlapping.json" ) )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll( "Mapper for [_source] conflicts with existing mapper" );
+	}
+
+	private void setupAndUpdateIndex(String customMappingsFile) {
+		setupHelper.start()
+				.withSchemaManagement( StubMappingSchemaManagementStrategy.DROP_ON_SHUTDOWN_ONLY )
+				.withIndexProperty( index.name(), ElasticsearchIndexSettings.SCHEMA_MANAGEMENT_MAPPINGS_FILE,
+						"custom-index-mappings/" + customMappingsFile
+				)
+				.withIndex( index )
+				.setup();
+
+		Futures.unwrappedExceptionJoin( index.schemaManager().createOrUpdate() );
+	}
+}
```

Comment:
I believe tests (and implementation) of validation are missing?

To be clear, most of validation should already be here; you only need to add code to validate `AbstractTypeMapping#extraAttributes`. And, obviously, to add tests.

You can find an example of how to do that in `org.hibernate.search.backend.elasticsearch.validation.impl.IndexSettingsValidator#validate`. It's important to use `validateAllIgnoreUnexpected` here, because we don't care about extra mapping attributes.